Sources/Abstract/Wrapper.swift

```diff
@@ -0,0 +1,31 @@
+/*:
+# Wrapper
+
+And abstract definition for a type that wraps another, and guarantees both initialization with and access to the wrapped value.
+
+The requirement is that, by constructing a `Wrapper` with a certain `Wrapped`, we can retrieve exactly the same value, thus:
+
+Wrapped(object).unwrap == object
+
+In a sense it means that the function `Wrapper.init(_:)` is isomorphic: from an instance of `Wrapper` I can always get a `Wrapped` and viceversa.
```

Comment:
I believe the proper way to phrase this is that `Wrapper` and the `Wrapped` value are isomorphic and the `Wrapper.init(_:)` and the `unwrap` computed property are self-inverses